jdaviz/configs/default/plugins/plot_options/plot_options.py

```diff
@@ -625,9 +634,74 @@ def _update_stretch_histogram(self, msg={}):
             # in case only the sample has changed but its length has not,
             # we'll force the traitlet to trigger a change
             hist_mark.send_state('sample')
-
         self._stretch_histogram_needs_update = False
 
+    @observe('stretch_vmin_value', 'stretch_vmax_value', 'layer_selected',
+             'stretch_function_value', 'stretch_hist_nbins',
+             'image_contrast_value', 'image_bias_value')
+    def _update_stretch_curve(self, msg=None):
+        # Import here to prevent circular import (and not at the top of the method so the import
+        # check is avoided, whenever possible).
+        from jdaviz.configs.imviz.plugins.viewers import ImvizImageView
+        from jdaviz.configs.cubeviz.plugins.viewers import CubevizImageView
+
+        if (
+            not isinstance(self.viewer.selected_obj, (ImvizImageView, CubevizImageView)) or
+            not hasattr(self, 'stretch_histogram')
+        ):
+            # don't update histogram if selected viewer is not an image viewer,
+            # or the stretch histogram hasn't been initialized:
+            return

```

Comment:
I don't love that this is duplicated logic, maybe we should put this in a re-usable method/property?  Or even in `ViewerSelect` itself so we could do `self.viewer.selected_is_image_viewer`?

Reply:
Moved 👍🏻